packages/opentelemetry-node-tracer/test/NodeTracer.test.ts

```diff
@@ -26,6 +26,11 @@ import {
 import { AsyncHooksScopeManager } from '@opentelemetry/scope-async-hooks';
 import { NodeTracer } from '../src/NodeTracer';
 
+const sleep = (time: number) =>
+  new Promise(resolve => {
```

Comment:
I remember having a conversation with @rochdev about us not using Promises in the library early on at one of the SIGs, though maybe that only applies to the core / API / basic tracer packages.

Roch, how do you feel about a Promise here?

Reply:
Just curious, did that discussion apply to the entire library, tests included? I'm assuming the scope manager should work with promises, but this library shouldn't use any of them in implementations for perf,etc reasons?

Reply:
Yes, that makes total sense! I confess that I was looking through this code a bit faster than I should have been. Use of a Promise here to test the scope manager seems needed for sure.